packages/build/src/plugins/options.js

```diff
@@ -55,16 +55,16 @@ const getCorePluginInputs = function(corePlugin, plugins) {
 }
 
 // Optional core plugins requires user opt-in
-const isOptionalCore = function({ package, optional }, plugins) {
-  return optional && plugins.every(plugin => plugin.package !== package)
+const isOptionalCore = function(pluginA, plugins) {
+  return pluginA.optional && plugins.every(pluginB => pluginB.package !== pluginA.package)
 }
 
-const isUserPlugin = function({ package }) {
-  return !isCorePlugin(package)
+const isUserPlugin = function(plugin) {
+  return !isCorePlugin(plugin.package)
 }
 
-const normalizePluginOptions = function({ package, pluginPath, loadedFrom, origin, inputs }) {
-  return { package, pluginPath, loadedFrom, origin, inputs }
+const normalizePluginOptions = function({ package: packageName, pluginPath, loadedFrom, origin, inputs }) {
```

Comment:
From that point, the output of `@netlify/config` is normalized from `package` to `packageName`.